Improve cds-typer Documentation (#430)
cap-js/docs: tools/cds-typer.md

Using anything but lambda functions for either CRUD handler or action implementation will make it impossible for the LSP to infer the parameter types.

> You can remedy this by specifying the expected type yourself via [JSDoc](https://jsdoc.app/) in JavaScript projects:

Reviewer: ```suggestion
You can remedy this by specifying the expected type with one of the following options:

Using [JSDoc](https://jsdoc.app/) in JavaScript projects:
```

Using anything but lambda functions for either CRUD handler or action implementation will make it impossible for the LSP to infer the parameter types.

> You can remedy this by specifying the expected type with one of the following options:

Reviewer: ```suggestion
You can remedy this by specifying the expected type with one of the following options.
```

}
```

> or via `import type` in TypeScript projects:

Reviewer: ```suggestion
Using `import type` in TypeScript projects:
```
